draft-ietf-httpbis-message-signatures.md

```diff
@@ -1790,6 +1793,20 @@ From here, the signing process proceeds as usual.
 
 Upon verification, it is important that the verifier validate not only the signature but also the value of the Content-Digest field itself against the actual received content. Unless the verifier performs this step, it would be possible for an attacker to substitute the message content but leave the Content-Digest field value untouched. Since only the field value is covered by the signature directly, checking only the signature is not sufficient protection against such a substitution attack.
 
+## Non-List Field Values {#security-non-list}
+
+When an HTTP field occurs multiple times in a single message, these values need to be combined into a single one-line string value to be included in the HTTP signature base, as described in {{create-sig-input}}. Not all HTTP fields can be combined into a single value and still be a valid value for the field. However, for the purposes of generating the signature base, the message component value is never read back out of the signature base string and used in anyway. Therefore it is considered best practice to treat the signature base generation algorithm separately from processing the field values by the application, particularly for fields that are known to have this property.
+
+For example, the Set-Cookie field {{COOKIE}} defines an internal syntax that does not conform to the List syntax in {{STRUCTURED-FIELDS}}. This field is also typically sent as multiple fields with distinct values when sending multiple cookies. When multiple Set-Cookie fields are sent in the same message, it is not possible to combine these into a single and be able to parse and use the results, as discussed in {{HTTP, Section 5.3}}. Therefore, all the cookies need to be processed from their separate header values, without being combined, while the signature base needs to be processed from the special combined value generated solely for this purpose.
```

Comment:
This misses the attack.

Say you have:

```
Set-Cookie: that, happens, to, contain, a, comma
```

An attacker is now able to construct a variant message that produces the same signature:

```
Set-Cookie: that, happens
Set-Cookie: to, contain, a, comma
```

This will be processed differently, but it will be signed and treated as such.

For me, this is an unacceptable property of a signature scheme.

Reply:
Thanks, I didn't call out the commas specifically here but I'll tweak the text to address that.